src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/ManifestRecord.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+
+public abstract class ManifestRecord : IComparable<ManifestRecord>
+{
+    public string Path { get; set; } = null!;
```

Comment:
These `null!` everywhere are very much a mistake in my opinion. Either disable nullable in this file, or do it correctly. As it is this is lying about nullable, so will cause errors at runtime. It may be fine in this PR, but think about 3 years from now when you are fixing an issue and haven't touched this code in years.

Reply:
@MilenaHristova can you add a constructor where the argument names match the properties (apart from casing, so 'path' 'Path' for instance)?
I think `System.Text.Json` can figure out how to use this ctor when deserializing and we can then initialize the properties in the ctor.

Reply:
they can also be init only required properties if you prefer I think. S.T.Json can deserialize those if you aren't doing code gen.

Reply:
Oh, sweet, didn't know this works already, thanks!

@MilenaHristova so no ctor, just make them `public required string ..` with no default value.